// @fluentui/date-time-utilities/lib-amd/dateFormatting/dateFormatting.defaults.js
define(["require", "exports", "tslib"], function (require, exports, tslib_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /**
     * Format date to a day string representation
     * @param date - input date to format
     */
    exports.formatDay = function (date) { return date.getDate().toString(); };
    /**
     * Format date to a month-day-year string
     * @param date - input date to format
     * @param strings - localized strings
     */
    exports.formatMonthDayYear = function (date, strings) {
        return strings.months[date.getMonth()] + ' ' + date.getDate() + ', ' + date.getFullYear();
    };
    /**
     * Format date to a month-year string
     * @param date - input date to format
     * @param strings - localized strings
     */
    exports.formatMonthYear = function (date, strings) {
        return strings.months[date.getMonth()] + ' ' + date.getFullYear();
    };
    /**
     * Format date to a month string
     * @param date - input date to format
     * @param strings - localized strings
     */
    exports.formatMonth = function (date, strings) { return strings.months[date.getMonth()]; };
    /**
     * Format date to a year string representation
     * @param date - input date to format
     */
    exports.formatYear = function (date) { return date.getFullYear().toString(); };
    exports.DEFAULT_DATE_GRID_STRINGS = {
        months: [
            'January',
            'February',
            'March',
            'April',
            'May',
            'June',
            'July',
            'August',
            'September',
            'October',
            'November',
            'December',
        ],
        shortMonths: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
        days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
        shortDays: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
    };
    exports.DEFAULT_DATE_FORMATTING = {
        formatDay: exports.formatDay,
        formatMonth: exports.formatMonth,
        formatYear: exports.formatYear,
        formatMonthDayYear: exports.formatMonthDayYear,
        formatMonthYear: exports.formatMonthYear,
    };
    exports.DEFAULT_CALENDAR_STRINGS = tslib_1.__assign(tslib_1.__assign({}, exports.DEFAULT_DATE_GRID_STRINGS), { goToToday: 'Go to today', weekNumberFormatString: 'Week number {0}', prevMonthAriaLabel: 'Previous month', nextMonthAriaLabel: 'Next month', prevYearAriaLabel: 'Previous year', nextYearAriaLabel: 'Next year', prevYearRangeAriaLabel: 'Previous year range', nextYearRangeAriaLabel: 'Next year range', closeButtonAriaLabel: 'Close', selectedDateFormatString: 'Selected date {0}', todayDateFormatString: "Today's date {0}", monthPickerHeaderAriaLabel: '{0}, change year', yearPickerHeaderAriaLabel: '{0}, change month', dayMarkedAriaLabel: 'marked' });
});
